IdentityCore/src/cache/MSIDTokenCacheDataSource.h

```diff
@@ -33,6 +33,11 @@
 
 @protocol MSIDTokenCacheDataSource <NSObject>
 
+@optional
+@property (readonly) id accessControlForSharedItems;
```

Comment:
seems very macOS specific, does it need to be here? This is a common interface.

Reply:
Removed this logic and used dataSource casting to MSIDMacKeychainTokenCache only when we are sure it is of the same type.Please take a look.